Formula/julia.rb

```diff
@@ -21,6 +21,7 @@ class Julia < Formula
   end
 
   depends_on "cmake" => :build # Needed to build LLVM
+  depends_on "suite-sparse" => :test # Check bundled copy is used
```

Comment:
Looks like this should cause test to fail if it picks Homebrew's copy, e.g.
```
ERROR: InitError: could not load library "@rpath/libamd.2.dylib"
dlopen(@rpath/libamd.2.dylib, 0x0001): Symbol not found: _SuiteSparse_config
  Referenced from: <FD11A6EC-BC03-3888-B1B0-688297C2A999> /usr/local/Cellar/julia/1.8.5/lib/julia/libamd.2.4.6.dylib
  Expected in:     <B483234B-A332-3799-89B2-51555A21A0C4> /usr/local/Cellar/suite-sparse/7.0.1/lib/libsuitesparseconfig.7.0.1.dylib
```

Though, if `suite-sparse` is installed during build time then it causes issues as it looks like some executables will pick up Homebrew's copy. Shouldn't cause an issue unless we need to rebuild `julia` in same PR as `suite-sparse` (unlikely unless some common dependency).